test/integration/local/test_training.py

```diff
@@ -36,38 +40,78 @@ def test_py_versions(docker_image, processor, py_full_version):
     assert os.path.exists(os.path.join(output_path, 'my_model.h5')), 'model file not found'
 
 
 @pytest.mark.skip_cpu
 def test_gpu(sagemaker_local_session, docker_image):
-    resource_path = os.path.join(os.path.dirname(__file__), '../..', 'resources')
-    run_tf_single_training(script=os.path.join(resource_path, 'gpu_device_placement.py'),
-                           instance_type='local_gpu',
-                           sagemaker_local_session=sagemaker_local_session,
-                           docker_image=docker_image,
-                           training_data_path='file://{}'.format(
-                               os.path.join(resource_path, 'mnist', 'data')))
-
-
-def run_tf_single_training(script, instance_type, sagemaker_local_session,
-                           docker_image, training_data_path):
-    estimator = TensorFlow(entry_point=script,
-                           role='SageMakerRole',
-                           training_steps=1,
-                           evaluation_steps=1,
-                           train_instance_count=1,
-                           train_instance_type=instance_type,
-                           sagemaker_session=sagemaker_local_session,
-                           image_name=docker_image,
-                           base_job_name='test-tf-single')
+    run_tf_training(script=os.path.join(RESOURCE_PATH, 'gpu_device_placement.py'),
+                    instance_type='local_gpu',
+                    instance_count=1,
+                    sagemaker_local_session=sagemaker_local_session,
+                    docker_image=docker_image,
+                    training_data_path='file://{}'.format(
+                        os.path.join(RESOURCE_PATH, 'mnist', 'data')))
+
+
+@pytest.mark.skip_gpu
+def test_distributed_training_cpu(sagemaker_local_session, docker_image):
+    run_tf_training(script=os.path.join(RESOURCE_PATH, 'mnist', 'distributed_mnist.py'),
+                    instance_type='local',
+                    instance_count=2,
+                    sagemaker_local_session=sagemaker_local_session,
+                    docker_image=docker_image,
+                    training_data_path='file://{}'.format(
+                        os.path.join(RESOURCE_PATH, 'mnist', 'data-distributed')))
+
+
+@pytest.mark.skip_gpu
+@pytest.mark.parametrize('ps_num', [1, 2])
+def test_distributed_training_cpu_ps(sagemaker_local_session, docker_image, ps_num):
+    run_tf_training(script=os.path.join(RESOURCE_PATH, 'mnist', 'distributed_mnist.py'),
+                    instance_type='local',
+                    instance_count=2,
+                    sagemaker_local_session=sagemaker_local_session,
+                    docker_image=docker_image,
+                    hyperparameters={'sagemaker_parameter_server_num': ps_num},
+                    training_data_path='file://{}'.format(
+                        os.path.join(RESOURCE_PATH, 'mnist', 'data-distributed')))
+
+
+class ScriptModeTensorFlow(Framework):
```

Comment:
why subclass `Framework` and not `TensorFlow`?

Reply:
Trying to get rid off some of the unnecessary constructor args from TensorFlow.